components/org.wso2.carbon.identity.api.server.flow.management/org.wso2.carbon.identity.api.server.flow.management.v1/src/main/java/org/wso2/carbon/identity/api/server/flow/management/v1/response/handlers/PasswordRecoveryFlowMetaHandler.java

```diff
@@ -0,0 +1,107 @@
+import static org.wso2.carbon.identity.api.server.flow.management.v1.constants.FlowEndpointConstants.MAGIC_LINK_EXECUTOR;
+import static org.wso2.carbon.identity.api.server.flow.management.v1.constants.FlowEndpointConstants.PASSWORD_PROVISIONING_EXECUTOR;
+import static org.wso2.carbon.identity.api.server.flow.management.v1.constants.FlowEndpointConstants.SMS_OTP_EXECUTOR;
+import static org.wso2.carbon.identity.api.server.flow.management.v1.constants.FlowEndpointConstants.USER_RESOLVE_EXECUTOR;
+import static org.wso2.carbon.identity.recovery.IdentityRecoveryConstants.ConnectorConfig.PASSWORD_RECOVERY_EMAIL_LINK_ENABLE;
+import static org.wso2.carbon.identity.recovery.IdentityRecoveryConstants.ConnectorConfig.PASSWORD_RECOVERY_SEND_OTP_IN_EMAIL;
+import static org.wso2.carbon.identity.recovery.IdentityRecoveryConstants.ConnectorConfig.PASSWORD_RECOVERY_SMS_OTP_ENABLE;
+
+/**
+ * Handler for managing the password recovery flow meta information.
+ * This class extends the AbstractMetaResponseHandler to provide specific
+ * implementations for password recovery flow.
+ */
+public class PasswordRecoveryFlowMetaHandler extends AbstractMetaResponseHandler {
+
+    @Override
+    public String getFlowType() {
+
+        return String.valueOf(PASSWORD_RECOVERY);
+    }
+
+    @Override
+    public String getAttributeProfile() {
+
+        return END_USER_ATTRIBUTE_PROFILE;
+    }
+
+    @Override
+    public PasswordRecoveryConnectorConfigs getConnectorConfigs() {
+
+        Utils utils = new Utils();
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        PasswordRecoveryConnectorConfigs connectorConfigs = new PasswordRecoveryConnectorConfigs();
+        connectorConfigs.setMultiAttributeLoginEnabled(
+                utils.isFlowConfigEnabled(tenantDomain,
+                        MultiAttributeLoginConstants.MULTI_ATTRIBUTE_LOGIN_PROPERTY));
+        connectorConfigs.setPasswordRecoveryEnabled(
+                utils.isFlowConfigEnabled(tenantDomain,
+                        PASSWORD_RECOVERY_EMAIL_LINK_ENABLE) ||
+                utils.isFlowConfigEnabled(tenantDomain,
+                        PASSWORD_RECOVERY_SMS_OTP_ENABLE) ||
+                utils.isFlowConfigEnabled(tenantDomain,
+                        PASSWORD_RECOVERY_SEND_OTP_IN_EMAIL));
+
+        return connectorConfigs;
+    }
+
+    @Override
+    public List<String> getRequiredInputFields() {
+
+        return new ArrayList<>(getLoginInputFields());
+    }
+
+    @Override
+    public PasswordRecoveryFlowMetaResponse createResponse() {
+
+        PasswordRecoveryFlowMetaResponse response = new PasswordRecoveryFlowMetaResponse();
+        response.setFlowType(getFlowType());
+        response.setAttributeProfile(getAttributeProfile());
+        response.setSupportedExecutors(getSupportedExecutors());
+        response.setConnectorConfigs(getConnectorConfigs());
+        return response;
+    }
```

Comment:
I think we have the same implementation here similar to the abstract class. Ideally, this should work without overriding this method. Could you please verify?